### DISCUSSION TITLE
balance/balance4-principle

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#169] [General]
# balance/balance4-principle

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/balance/balance4-principle.html

**Comment by anaer** [comment]
`也称为“单臂模式”（Single Legged Model），如图 4-2 所示。`  图编号错误 4-7

**Reply by isno** [reply]
谢谢，已修正。

**Reply by anaer** [reply]
`图4-7 链路层 DSA 模式负载均衡` DSR吧

**Reply by isno** [reply]
:'( 是的，已修正。

**Comment by fFee-ops** [comment]
图 4-10 NAT 模式负载均衡中 从负载均衡器分发请求到realServer的时候是不是应该为Source[ip2,port2]，请求包的源地址应该是负载均衡器的IP

**Reply by isno** [reply]
是的，谢谢。已修改

**Comment by linjiaqin12138** [comment]
Tunnel 模式中，真实服务器和LoadBalancer都是相同的IP:端口，而且都面对外网，client发送请求的时候，IP数据包为什么一定会打到LoadBalancer上呢？

**Reply by isno** [reply]
因为它是 balancer，所以请求先打到 balancer 🫡

其次，ip 是 vip。真实服务器只是 lo 绑定的这个 vip

**Comment by xiongyunhua-star** [comment]
它会保持原始请求中的“源 IP 地址”，即虚拟 IP（1.1.1.1），作为响应数据包的源 IP。

-----> 应该是---->

它会保持原始请求中的"目的IP 地址”，即虚拟 IP（1.1.1.1），作为响应数据包的源 IP  ????

**Reply by isno** [reply]
对，这里我写错了。已修改，谢谢。
